python/taichi/lang/matrix_ops.py

```diff
@@ -0,0 +1,39 @@
+from taichi.lang.impl import static
+from taichi.lang.kernel_impl import func, pyfunc
+from taichi.lang.matrix_ops_utils import (arg_at, assert_tensor, preconditions,
+                                          square_matrix)
+from taichi.lang.ops import cast
+from taichi.lang.util import in_taichi_scope
+from taichi.types.annotations import Template
+
+
+@preconditions(square_matrix)
+@pyfunc
+def trace(x):
+    shape = static(x.get_shape())
+    result = cast(0, x.element_type()) if static(in_taichi_scope()) else 0
+    # TODO: get rid of static when
+    # CHI IR Tensor repr is ready stable
+    for i in static(range(shape[0])):
+        result += x[i, i]
+    return result
+
+
+@preconditions(arg_at(0, assert_tensor))
+@func
+def fill(m: Template(), val):
+    s = static(m.get_shape())
```

Comment:
super nit: the naming seems a little inconsistent. `trace` uses `x` for the matrix and `shape` for the shape, while `fill` uses `m` for the matrix and `s`  for the shape. Shall we unify them so that future developers can easily find a convention to follow? I suggest using `mat` and `shape`.  WDYT

Reply:
Sounds good. Let's use `mat` for matrices, `vec` for vectors and `shape` for shapes.